lib/src/front_end/piece_factory.dart

```diff
@@ -526,22 +526,34 @@ mixin PieceFactory implements CommentWriter {
     if (body.keyword != null) space();
   }
 
-  /// Creates a [Piece] for some code followed by an `=` and an expression in
-  /// any place where an `=` appears:
+  /// Creates a [Piece] with "assignment-like" splitting.
+  ///
+  /// This is used, obviously, for assignments and variable declarations to
+  /// handle splitting after the `=`, but is also used in any context where an
+  /// expression follows something that it "defines" or "initializes":
   ///
   /// * Assignment
   /// * Variable declaration
   /// * Constructor initializer
+  /// * Expression (`=>`) function body
+  /// * Named argument or named record field (`:`)
+  /// * Map entry (`:`)
+  /// * For-in loop iterator (`in`)
   ///
-  /// This is also used for map literal entries and named arguments which are
-  /// also sort of like bindings. In that case, [operator] is the `:`.
-  ///
-  /// This method assumes the code to the left of the `=` or `:` has already
+  /// This method assumes the code to the left of the operator has already
   /// been visited.
-  void finishAssignment(Token operator, Expression rightHandSide) {
-    if (operator.type == TokenType.EQ) space();
-    token(operator);
-    var target = pieces.split();
+  void finishAssignment(Token operator, Expression rightHandSide,
+      {bool splitBeforeOperator = false}) {
```

Comment:
[nit] can you describe the new argument in the doc?

Reply:
Good catch, done.